dm/config/task.go

```diff
@@ -585,6 +585,9 @@ func (c *TaskConfig) adjust() error {
 	if c.TaskMode != ModeFull && c.TaskMode != ModeIncrement && c.TaskMode != ModeAll {
 		return terror.ErrConfigInvalidTaskMode.Generate()
 	}
+	if c.MetaSchema == "" {
```

Comment:
Do we need copy or refactor the default value here for engine? https://github.com/pingcap/tiflow/blob/de8a5068e3cbb169b3ab18acbb4e278046712b96/dm/config/task.go#L483

Reply:
JobCfg.adjust calls https://github.com/pingcap/tiflow/blob/5bd84aea209e9f962e080d2fb2a005a79773591e/engine/jobmaster/dm/config/config.go#L187-L189

we don't use NewTaskConfig now. Will check it soon

Reply:
> Anyway, assigning default value in adjust is more safe in case we forget to use NewTaskConfig in future

Reply:
> Anyway, assigning default value in adjust is more safe in case we forget to use NewTaskConfig in future

Agreed. What I mean is that did you check all default variable in NewTaskConfig is now copy to adjust? Such as CleanDumpFiles, OnlineDDL...

Reply:
we can't tell the default value and zero value. Will use NewTaskConfig in toDMTaskConfig instead.